packages/ckb-indexer/src/transactionCollector.ts

```diff
@@ -0,0 +1,442 @@
+          params: [hashItem.tx_hash],
+        };
+      }
+    );
+    const transactionList: TransactionWithIO[] = await requestBatch(
+      this.CKBRpcUrl,
+      getDetailRequestData
+    ).then((response: GetTransactionRPCResult[]) => {
+      return response.map(
+        (item: GetTransactionRPCResult): TransactionWithIO => {
+          if (!this.filterOptions.skipMissing && !item.result) {
+            throw new Error(
+              `Transaction ${
+                transactionHashList.objects[item.id].tx_hash
+              } is missing!`
+            );
+          }
+          const ioType = transactionHashList.objects[item.id].io_type;
+          const ioIndex = transactionHashList.objects[item.id].io_index;
+          return { ioType, ioIndex, ...item.result };
+        }
+      );
+    });
+    return transactionList;
+  };
+
+  private getCellByOutpoint = async (output: OutPoint) => {
+    const transaction: Transaction = await request(
+      this.CKBRpcUrl,
+      "get_transaction",
+      [output.tx_hash]
+    );
+    return transaction.outputs[parseInt(output.index)];
+  };
+
+  private isCellScriptArgsValidate = (targetCell: Output) => {
+    let lockArgsLen: number | "any" | undefined;
+    let typeArgsLen: number | "any" | undefined;
+    if (this.queries.lock) {
+      lockArgsLen = instanceOfScriptWrapper(this.queries.lock)
+        ? this.queries.lock.argsLen
+        : this.queries.argsLen;
+    }
+    if (this.queries.type && this.queries.type !== "empty") {
+      typeArgsLen = instanceOfScriptWrapper(this.queries.type)
+        ? this.queries.type.argsLen
+        : this.queries.argsLen;
+    }
+    const resultMap: cellFilterResult = {
+      lockArgsLen: true,
+      typeArgsLen: true,
+      isTypeScriptEmpty: true,
+    };
+    if (this.queries.lock && instanceOfScriptWrapper(this.queries.lock)) {
+      resultMap.lockArgsLen = false;
+      if (getHexStringBytes(targetCell.lock.args) !== lockArgsLen) {
+        resultMap.lockArgsLen = true;
+      }
+    }
+    if (this.queries.type && this.queries.type !== "empty") {
+      resultMap.typeArgsLen = false;
+      if (
+        targetCell.type &&
+        getHexStringBytes(targetCell.type.args) !== typeArgsLen
+      ) {
+        resultMap.typeArgsLen = true;
+      }
+    }
+    if (this.queries.type && this.queries.type === "empty") {
+      resultMap.isTypeScriptEmpty = false;
+      if (!targetCell.type) {
+        resultMap.isTypeScriptEmpty = true;
+      }
+    }
+    return (
+      resultMap.lockArgsLen &&
+      resultMap.typeArgsLen &&
+      resultMap.isTypeScriptEmpty
+    );
+  };
+
+  private filterByIoType = (
+    inputResult: GetTransactionsResult[],
+    ioType: IOType
+  ) => {
+    if (ioType === "both") {
+      return inputResult;
+    }
+    if (ioType === "input" || ioType === "output") {
+      return inputResult.filter(
+        (item: GetTransactionsResult) =>
+          item.io_type === ioType || item.io_type === "both"
```

Comment:
Is it possible for `both` to appear in `GetTransactionsResult`?

Reply:
Yes,just like lastCursor, I change io_type to both if the transaction was find by both type script and lock script.